quadratic_simple_constraint.py

```diff
@@ -0,0 +1,49 @@
+"""
+Optuna example that optimizes a simple quadratic function with constraints.
+
+In this example, we optimize a simple quadratic function with constraints.
+
+Please check https://optuna.readthedocs.io/en/stable/faq.html#id16 as well.
+
+"""
+
+import optuna
+
+
+# Define a simple 2-dimensional objective function with constraints.
+def objective(trial):
+    x = trial.suggest_float("x", -100, 100)
+    y = trial.suggest_categorical("y", [-1, 0, 1])
+    return x**2 + y
+
+
+# Define a function that returns constraints.
+def constraints(trial):
+    params = trial.params
+    x, y = params["x"], params["y"]
+    # c1 <= 0 and c2 <= 0 must be satisfied.
+    c1 = 3 * x * y + 10
+    c2 = x * y + 30
+    return [c1, c2]
+
+
+if __name__ == "__main__":
+    # We minimize obj1 and maximize obj2.
+    sampler = optuna.samplers.NSGAIISampler(constraints_func=constraints)
```

Comment:
```suggestion
    sampler = optuna.samplers.TPESampler(constraints_func=constraints)
```
I believe the `NSGAIISampler` is not the first choice for single objective optimization. How about using `TPESampler` instead.

Reply:
I do not have a strong opinion about the choice, so I will replace it.
Btw, this is CNSGA-II (not NSGA-II, just noting).